client/nginx_test.go

```diff
@@ -224,9 +235,227 @@ func TestStreamDetermineUpdates(t *testing.T) {
 	}}
 
 	for _, test := range tests {
-		toAdd, toDelete := determineStreamUpdates(test.updated, test.nginx)
+		toAdd, toDelete, _ := determineStreamUpdates(test.updated, test.nginx)
 		if !reflect.DeepEqual(toAdd, test.expectedToAdd) || !reflect.DeepEqual(toDelete, test.expectedToDelete) {
 			t.Errorf("determiteUpdates(%v, %v) = (%v, %v)", test.updated, test.nginx, toAdd, toDelete)
 		}
 	}
 }
+
+func TestDetermineUpdatesWithUpdate(t *testing.T) {
+	maxConns := 1
+	var tests = []struct {
+		updated          []UpstreamServer
+		nginx            []UpstreamServer
+		expectedToUpdate []UpstreamServer
+	}{{
+		updated: []UpstreamServer{
+			{
+				Server:   "10.0.0.1:80",
+				MaxConns: &maxConns,
+			},
+		},
+		nginx: []UpstreamServer{
+			{
+				Server: "10.0.0.1:80",
+			},
+			{
+				ID:     2,
+				Server: "10.0.0.2:80",
+			},
+		},
+		expectedToUpdate: []UpstreamServer{
+			{
+				Server:   "10.0.0.1:80",
+				MaxConns: &maxConns,
+			},
+		}}}
+
+	for _, test := range tests {
+		_, _, toUpdate := determineUpdates(test.updated, test.nginx)
+		if !reflect.DeepEqual(toUpdate, test.expectedToUpdate) {
+			t.Errorf("determiteUpdates(%v, %v) = (%v)", test.updated, test.nginx, toUpdate)
+		}
+	}
+}
+
+func TestStreamDetermineUpdatesWithUpdate(t *testing.T) {
+	maxConns := 1
+	var tests = []struct {
+		updated          []StreamUpstreamServer
+		nginx            []StreamUpstreamServer
+		expectedToUpdate StreamUpstreamServer
+	}{{
+		updated: []StreamUpstreamServer{
+			{
+				Server:   "10.0.0.1:80",
+				MaxConns: &maxConns,
+			},
+		},
+		nginx: []StreamUpstreamServer{
+			{
+				Server: "10.0.0.1:80",
+			},
+			{
+				ID:     2,
+				Server: "10.0.0.2:80",
+			},
+		},
+		expectedToUpdate: StreamUpstreamServer{
+			Server:   "10.0.0.1:80",
+			MaxConns: &maxConns,
+		}}}
+
+	for _, test := range tests {
+		_, _, toUpdate := determineStreamUpdates(test.updated, test.nginx)
+
+		for _, update := range toUpdate {
+			if !reflect.DeepEqual(update, test.expectedToUpdate) {
```

Comment:
this loop looks unnecessary.

Please consider the following case with `10.0.0.3` added. I think it makes sense that determine*Updates handles that:
```go
func TestStreamDetermineUpdatesWithUpdate(t *testing.T) {
	maxConns := 1
	var tests = []struct {
		updated          []StreamUpstreamServer
		nginx            []StreamUpstreamServer
		expectedToUpdate []StreamUpstreamServer
	}{{
		updated: []StreamUpstreamServer{
			{
				Server:   "10.0.0.1:80",
				MaxConns: &maxConns,
			},
			{
				Server:   "10.0.0.3",
				MaxConns: &maxConns,
			},
		},
		nginx: []StreamUpstreamServer{
			{
				Server: "10.0.0.1:80",
			},
			{
				ID:     2,
				Server: "10.0.0.2:80",
			},
			{
				Server: "10.0.0.3:80",
			},
		},
		expectedToUpdate: []StreamUpstreamServer{
			{
				Server:   "10.0.0.1:80",
				MaxConns: &maxConns,
			},
			{
				Server:   "10.0.0.3",
				MaxConns: &maxConns,
			},
		}}}

	for _, test := range tests {
		_, _, toUpdate := determineStreamUpdates(test.updated, test.nginx)

		if !reflect.DeepEqual(toUpdate, test.expectedToUpdate) {
			t.Errorf("determiteUpdates(%v, %v) = (%v)", test.updated, test.nginx, toUpdate)
		}
	}
}
```